### PR TITLE
Refactored the Video widget to use the Youtube Player API...

### PR DESCRIPTION
**Pull request by udidol**
instead of a simple iframe embed

PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Video widget - Changed Youtube videos to be rendered and loaded by the YouTube Player API instead of a basic iframe

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: assets/dev/js/frontend/handlers/video.js [hunk 1]
```diff
@@ -27,6 +27,15 @@ class VideoModule extends elementorModules.frontend.handlers.Base {
 		if ( ! this.getElementSettings( 'lightbox' ) ) {
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/video.js:27]: You should handle all cases, including lightbox

File: includes/widgets/video.php [hunk 2]
```diff
@@ -243,9 +245,7 @@ protected function _register_controls() {
 				'label' => __( 'Start Time', 'elementor' ),
 				'type' => Controls_Manager::NUMBER,
 				'description' => __( 'Specify a start time (in seconds)', 'elementor' ),
-				'condition' => [
```
Comment (kobizz) [inline includes/widgets/video.php:246]: Is the loop option getting along well with the start and end parameters?

File: includes/widgets/video.php [hunk 3]
```diff
@@ -822,7 +830,11 @@ protected function render() {
 		<div <?php echo $this->get_render_attribute_string( 'video-wrapper' ); ?>>
 			<?php
 			if ( ! $settings['lightbox'] ) {
-				echo $video_html; // XSS ok.
+				if ( 'youtube' === $settings['video_type'] ) {
```
Comment (kobizz) [inline includes/widgets/video.php:833]: You shouldn't fetch the video html when it's a YouTube video

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/widgets/video.php (hunk 2 above)
Comment (udidol) [reply includes/widgets/video.php:246]: From my tests yes but we could give QA to test

File: includes/widgets/video.php (hunk 3 above)
Comment (udidol) [reply includes/widgets/video.php:840]: don't understand

File: assets/dev/js/frontend/handlers/video.js (hunk 1 above)
Comment (udidol) [reply assets/dev/js/frontend/handlers/video.js:27]: But the Lightbox module handles its own videos

File: assets/dev/js/frontend/handlers/video.js [hunk 4]
```diff
@@ -66,10 +75,81 @@ class VideoModule extends elementorModules.frontend.handlers.Base {
...
+		super.onInit();
+
+		const elementSettings = this.getElementSettings();
+
+		if ( 'youtube' === elementSettings.video_type ) {
+			this.apiProvider = elementorFrontend.utils.youtube;
+		} else {
+			// Currently the only API integration in the Video widget is for the YT API
+			return;
+		}
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/video.js:139]:
```suggestion
		if ( 'youtube' !== elementSettings.video_type ) {
			// Currently the only API integration in the Video widget is for the YT API
			return;
		}

		this.apiProvider = elementorFrontend.utils.youtube;
```

File: assets/dev/js/frontend/handlers/video.js [hunk 5]
```diff
@@ -66,10 +75,81 @@ class VideoModule extends elementorModules.frontend.handlers.Base {
...
+		this.videoID = this.apiProvider.getVideoIDFromURL( elementSettings.youtube_url );
+
+		// If there is an image overlay, the YouTube video prep method will be triggered on click
+		if ( ! this.videoID || elementSettings.show_image_overlay ) {
+			return;
+		}
+
+		this.apiProvider.onApiReady( ( apiObject ) => {
+			this.prepareYTVideo( apiObject );
+		} );
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/video.js:150]:
```suggestion
		this.apiProvider.onApiReady( ( apiObject ) => this.prepareYTVideo( apiObject ) );
```

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/widgets/video.php (hunk 3 above)
Comment (kobizz) [reply includes/widgets/video.php:842]: Do the condition above

File: assets/dev/js/frontend/handlers/video.js (hunk 5 above)
Comment (udidol) [reply assets/dev/js/frontend/handlers/video.js:150]: Done.

File: assets/dev/js/frontend/handlers/video.js (hunk 4 above)
Comment (udidol) [reply assets/dev/js/frontend/handlers/video.js:139]: Done.

**Comment by rodolphebertozzo** [comment]
Hi, can you take a look at this issue which is problematic and deals with loading YouTube videos as wallpaper?

https://github.com/elementor/elementor/issues/12198

**Review by kobizz — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**